app/models/poll_vote.rb

```diff
@@ -6,21 +6,21 @@ class PollVote < ApplicationRecord
   counter_culture :poll_option
   counter_culture :poll
 
-  validates :poll_id, presence: true, presence: true, uniqueness: { scope: :user_id } # In the future we'll remove this constraint if/when we allow multi-answer polls
+  validates :poll_id, presence: true, uniqueness: { scope: :user_id } # In the future we'll remove this constraint if/when we allow multi-answer polls
   validates :poll_option_id, presence: true, uniqueness: { scope: :user_id }
   validate :one_vote_per_poll_per_user
 
   after_save :touch_poll_votes_count
   after_destroy :touch_poll_votes_count
 
-  def poll
-    poll_option.poll
-  end
+  delegate :poll, to: :poll_option
 
   private
 
   def one_vote_per_poll_per_user
-    errors.add(:base, "cannot vote more than once in one poll") if poll.poll_votes.where(user_id: user_id).any? || poll.poll_skips.where(user_id: user_id).any?
+    has_votes = (
+      poll.poll_votes.where(user_id: user_id).any? || poll.poll_skips.where(user_id: user_id).any?)
```

Comment:
When we have such validation if we do smth like `PollVote.new.valid` it'll fail with `NoMethodError: undefined method `poll' for nil:NilClass` which I'm not 100% comfortable with. Probably, we should execute this validation only when a `poll` exists.

Reply:
great catch! I'll fix it

Reply:
great catch! I'll fix it